basic/src/test/java/org/jboss/hal/testsuite/test/configuration/messaging/jmsbridge/JMSBridgeContextTestCase.java

```diff
@@ -0,0 +1,218 @@
+
+    @Drone
+    private WebDriver browser;
+
+    @Page
+    private JMSBridgePage page;
+
+    @BeforeClass
+    public static void beforeClass() throws CommandFailedException {
+        client.apply(new AddQueue.Builder(IN_QUEUE_NAME)
+                .jndiEntries(Collections.singletonList(IN_QUEUE_JNDI))
+                .replaceExisting()
+                .build());
+        client.apply(new AddQueue.Builder(OUT_QUEUE_NAME)
+                .jndiEntries(Collections.singletonList(OUT_QUEUE_JNDI))
+                .replaceExisting()
+                .build());
+
+        client.apply(new AddJMSBridge.Builder(JMS_BRIDGE_NAME)
+                .failureRetryInterval(1000)
+                .maxBatchSize(1000)
+                .maxBatchTime(1000)
+                .maxRetries(1)
+                .qualityOfService(AddJMSBridge.QualityOfService.AT_MOST_ONCE)
+                .sourceConnectionFactory(LOCAL_CONNECTION_FACTORY)
+                .sourceContext(Values.empty())
+                .sourceDestination(IN_QUEUE_JNDI)
+                .targetConnectionFactory(LOCAL_CONNECTION_FACTORY)
+                .targetContext(Values.empty())
+                .targetDestination(OUT_QUEUE_JNDI)
+                .replaceExisting()
+                .build()
+        );
+    }
+
+    @AfterClass
+    public static void afterClass() throws CommandFailedException, IOException, TimeoutException, InterruptedException {
+        try {
+            operations.remove(JMS_BRIDGE_ADDRESS);
+            client.apply(new RemoveQueue(IN_QUEUE_NAME));
+            client.apply(new RemoveQueue(OUT_QUEUE_NAME));
+            administration.reloadIfRequired();
+        } finally {
+            client.close();
+        }
+    }
+
+    @Test
+    public void addSourceContextProperty() throws Exception {
+        page.navigate();
+
+        Console.withBrowser(browser).dismissReloadRequiredWindowIfPresent();
+
+        page.selectJMSBridgeInTable(JMS_BRIDGE_NAME);
+        page.switchToSourceContext();
+
+        WizardWindow wizard = page.openAddContextPropertyWindow();
+
+        Editor editor = wizard.getEditor();
+
+        editor.text("key", SOURCE_KEY_TBA);
+        editor.text("value", SOURCE_VALUE_TBA);
+        boolean result = wizard.finishAndDismissReloadRequiredWindow();
+
+        assertTrue("Window should be closed", result);
+
+        assertTrue("Attribute '" + SOURCE_CONTEXT_IDENTIFIER + "' should have defined property in list: <'" +
+                        SOURCE_KEY_TBA + "','" + SOURCE_VALUE_TBA + ">.",
+                hasListAttributeGivenProperty(SOURCE_CONTEXT_IDENTIFIER, SOURCE_KEY_TBA, SOURCE_VALUE_TBA));
+    }
+
+    @Test
+    public void removeSourceContextProperty() throws Exception {
+        operations.writeAttribute(JMS_BRIDGE_ADDRESS,
+                SOURCE_CONTEXT_IDENTIFIER,
+                                    new ModelNode().add(SOURCE_KEY_TBR, SOURCE_VALUE_TBR))
```

Comment:
is the indentation correct, I would expect this line to have the same indentation as the one above.